### PR TITLE
Feature/ci cd workflow

### PR DESCRIPTION
**Pull request by pinakinkansara**
- added CI CD work flow 
- rebased from develop
- code refactoring

### CODE REVIEW

**Comment by pinakinkansara** [comment]
@will-freshworks, please Check workflow-related code.
CI/CD triggers on each pull request and you can find artifacts in the actions tab under particular successful build.

Thinking of to add upload to play store in next ticked as need to discuss on few things.

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# name of the work flow
+name: Build
+on: [pull_request]
+jobs:
+  build:
+    if: ${{ !contains(github.event.head_commit.message, 'ci skip') }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest]
```
Comment (will-freshworks) [inline .github/workflows/build.yaml:10]: Do we really need mac-os? Surely we can run this build on ubuntu.

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# name of the work flow
+name: Build
+on: [pull_request]
```
Comment (will-freshworks) [inline .github/workflows/build.yaml:3]: Do we want to create a release for every action on every PR? I think this should be:
```
on:
  push:
    branches:
      - main
```

File: app/build.gradle [hunk 3]
```diff
@@ -14,8 +14,8 @@ android {
         applicationId "ca.bc.gov.vaxcheck"
         minSdk 27
         targetSdk 31
-        versionCode 11
-        versionName "1.0"
+        versionCode 12
```
Comment (will-freshworks) [inline app/build.gradle:17]: I feel like we're gonna run into problems if we can't auto-increment this on each build. How hard would this be to add?

**Review by will-freshworks — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yaml (hunk 1 above)
Comment (pinakinkansara) [reply .github/workflows/build.yaml:10]: Yes, Will, we can do it on Ubuntu and windows as well.

File: .github/workflows/build.yaml (hunk 2 above)
Comment (pinakinkansara) [reply .github/workflows/build.yaml:3]: Yes, I am also thinking on the same line.

File: app/build.gradle (hunk 3 above)
Comment (pinakinkansara) [reply app/build.gradle:17]: It's not difficult at all. The only thing I see is if we do auto-increment then we will lose control of the version name and code we publish on the play store.

**Pull request closed without merging**